contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpHandler.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.opencensus.trace.MessageEvent;
+import io.opencensus.trace.MessageEvent.Type;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.propagation.TextFormat;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.annotation.Nullable;
+
+/**
+ * A helper class to handle instrumentation in HTTP client/server.
+ *
+ * <p>This class provides basic operations, such as recording {@link MessageEvent} and handling
+ * errors.
+ *
+ * @since 0.13
+ */
+public abstract class HttpHandler<Q, P> {
+
+  /** The Open Census tracing component. */
+  Tracer tracer;
```

Comment:
Why not private final? Same for all the other members. If for test please use the annotation VisibleForTesting